tests/FunctionCallTest.php

```diff
@@ -433,52 +433,63 @@ public static function Baz($mixed) : string {
                      */
                     $elements = explode(" ", $string, $limit);',
                 'assertions' => [
-                    '$elements' => 'list<string>',
+                    '$elements' => 'array{0?: string, 1?: string, 2?: string, ...<int<0, max>, string>}',
```

Comment:
This is kinda ugly: the array type changed from `list<string>` to a weird union type here, possibly due to the fact that psalm performs a union type between `array<never, never>|array{string}|array{0: string, 1?: string}|non-empty-list<string>`.

This is a bit of a weird one, and unsure if it indicates a problem with this approach

Reply:
Related: https://github.com/vimeo/psalm/pull/9016#discussion_r1058458616

Reply:
c0c01168091ef8a2eadf8b06532f303b480d6686